### DISCUSSION TITLE
Outline does not print elements specified in target

### DISCUSSION THREAD

**Discussion by jmpunkt** [typst/typst#4392] [General]
## Issue
When trying to select certain elements (in this case sub-sub-sections) and put them into the outline target, then it does not print any elements.

## Expected

In the example below, I would expect that there is a TOC with the following elements: `B11, B111, B12, B121, B13, B131`. However, that is not the case, the TOC is empty without any elements. If we use query instead of outline, and print these elements, then all relevant element are printed correctly as an array. So the selector works and I would expect outline to work as well.

If only one part of the selectors is used (e.g. `selector(heading).after(elm.location(), inclusive: false)`), then the TOC is generated. But then also elements are included which should not be part of the TOC.

```typst
= A
== A1
=== A11
= B
== B1
#context {
  let before = query(
    selector(heading).before(here(), inclusive: true),
  )
  let elm = before.pop()
  let next = query(
    selector(heading).after(here(), inclusive: true),
  ).find(it => it.level <= elm.level)
  outline(
    target: selector(heading).after(elm.location(), inclusive: false).and(
      selector(heading).before(next.location(), inclusive: false),
    ),
  )
}
=== B11
==== B111
=== B12
==== B121
=== B13
==== B131
= B2
= C
```

## Used Version
0.11.1 on NixOS

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Your selector captures the `Contents` heading generated by the `outline`. If you set `title: none` in your `outline` call, it works.

**Reply by jmpunkt** [reply]
Thanks thats the hint I needed. I came up with a solution which works without setting the title to none as an alternative.

```typst
= Section 1
== Subsection 1
#context {
  let before = query(
    selector(heading).before(here(), inclusive: true),
  )
  let elm = before.pop()
  let after_elements = query(
    heading.where(outlined: true).after(here(), inclusive: true),
  )
  let next_maybe = after_elements.find(it => it.level <= elm.level)
  let next = if next_maybe == none {
    after_elements.pop()
  } else {
    next_maybe
  }
  outline(
    target: heading.where(outlined: true).after(
      elm.location(),
      inclusive: false,
    ).and(
      heading.where(outlined: true).before(
        next.location(),
        inclusive: next_maybe == none,
      ),
    ),
  )
}

=== Subsubsection 1
=== Subsubsection 2
```
